keanu-project/src/main/java/io/improbable/keanu/tensor/intgr/ScalarIntegerTensor.java

```diff
@@ -43,13 +43,13 @@ public int getRank() {
     }
 
     @Override
-    public long getLength() {
-        return isShapePlaceholder() ? 0L : 1L;
+    public long[] getStride() {
+        return new long[0];
```

Comment:
Keep it the same as `ScalarDoubleTensor`?

```
    public long[] getStride() {
        return Arrays.copyOf(shape, shape.length);
    }
```

Reply:
I've actually made a little hybrid of the two for both of them which to me is much more correct than either ;-)